### ISSUE TITLE
[REF-2444] IndexError: list index out of range when publishing custom component

### ISSUE THREAD

**Issue by masenf** [reflex-dev/reflex#2972]
**Describe the bug**
```
Publishing custom component to pypi...
Distribution files for the version to be published already exist. Do you want to rebuild? [n/y] (n): y
Building custom component...
Traceback (most recent call last):
  File "/Users/masen/code/reflex-dev/VENV-dev311/bin/reflex", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/typer/main.py", line 328, in __call__
    raise e
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/typer/main.py", line 311, in __call__
    return get_command(self)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/typer/core.py", line 778, in main
    return _main(
           ^^^^^^
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/typer/core.py", line 216, in _main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/masen/code/reflex-dev/VENV-dev311/lib/python3.11/site-packages/typer/main.py", line 683, in wrapper
    return callback(**use_params)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/masen/code/reflex-dev/reflex/reflex/custom_components/custom_components.py", line 666, in publish
    _ensure_dist_dir(version_to_publish=version_to_publish, build=build)
  File "/Users/masen/code/reflex-dev/reflex/reflex/custom_components/custom_components.py", line 573, in _ensure_dist_dir
    _run_build()
  File "/Users/masen/code/reflex-dev/reflex/reflex/custom_components/custom_components.py", line 437, in _run_build
    _make_pyi_files()
  File "/Users/masen/code/reflex-dev/reflex/reflex/custom_components/custom_components.py", line 421, in _make_pyi_files
    package_name = glob("custom_components/*.egg-info")[0].replace(".egg-info", "")
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

**To Reproduce**
I upgraded to 0.4.6a1 and attempted to publish a component. It failed.

**Specifics (please complete the following information):**
 - Python Version: 3.11.8
 - Reflex Version: 0.4.6a1
 - OS: macOS

<sub>[REF-2444](https://linear.app/reflex-dev/issue/REF-2444/indexerror-list-index-out-of-range-when-publishing-custom-component)</sub>

**Comment by wassafshahzad** [comment]
May I take it ?

**Comment by kshitiz305** [comment]
Hey @Lendemor may I solve this issue

**Comment by kshitiz305** [comment]
Issue has been solved in #3048
